packages/modelviewer.dev/examples/OOBE/JSInterface.js

```diff
@@ -0,0 +1,149 @@
+import copyData from './title_body_data.json' assert {type: 'json'};
+
+const COPY = copyData.copy;
+let prevTextId = 0;
+let currTextId = 0;
+
+const vignette = document.querySelector('#vignette');
+const card = document.querySelector('#card');
+const cardImg = document.querySelector('#card-image');
+const cardTitle = document.querySelector('#card-title');
+const cardBody = document.querySelector('#card-body');
+const dots = document.querySelector('#dots');
+const skipButton = document.querySelector('#skip');
+const nextButton = document.querySelector('#next');
+const backButton = document.querySelector('#back');
+
+// Display the correct UI for the given state
+function setStateParameters(state, calloutData) {
+    console.log("Displaying UI for: " + state.name);
+
+    updateTextCard(state.titleStringID, state.bodyStringID);
+    updateButtons(state);
+    updateDots(state.showDots, calloutData);
+}
+
+// Update text and image in the card
+function updateTextCard(titleStringID, bodyStringID) {
+    prevTextId = currTextId;
+    currTextId = titleStringID;
+
+    if (currTextId != prevTextId) {
+        animateCard();
+    }
+    
+    let data = findDataById(titleStringID, COPY);
+    if (data) {
+        // Update the image if available
+        if ('imageSrc'in data) {
+            cardImg.style.display = "block";
+            cardImg.src = data.imageSrc;
+        } else {
+            cardImg.style.display = "none";
+            cardImg.src = "";
+        }
+        cardTitle.innerHTML = data.titleString;
+        cardBody.innerHTML = data.bodyString;
+    }
+    else {
+        cardTitle.innerHTML = titleStringID;
+        cardBody.innerHTML = bodyStringID;
+    }
+}
+
+function animateCard() {
+    // Remove the animation class
+    card.classList.remove("run-slide-in-animation");
+    // Trigger reflow
+    void card.offsetWidth;
```

Comment:
I've never seen this JS syntax; what does this do?

Reply:
HTMLElement.offsetWidth is a read-only property that returns the layout width of an element as an integer. Calling it forces a reflow. Prob a better way to do this performance-wise.